sdk/iothub/src/service/mod.rs

```diff
@@ -268,10 +271,159 @@ impl ServiceClient {
             response_time_out,
         )
     }
+
+    /// Get the module twin of a given device and module
+    ///
+    /// ```
+    /// use iothub::service::ServiceClient;
+    ///
+    /// # let connection_string = "HostName=cool-iot-hub.azure-devices.net;SharedAccessKeyName=iothubowner;SharedAccessKey=YSB2ZXJ5IHNlY3VyZSBrZXkgaXMgaW1wb3J0YW50Cg==";
+    /// let iothub = ServiceClient::from_connection_string(connection_string, 3600).expect("Failed to create the ServiceClient!");
+    /// let twin = iothub.get_module_twin("some-device", "some-module");
+    /// ```
+    pub async fn get_module_twin<S, T>(
+        &self,
+        device_id: S,
+        module_id: T,
+    ) -> Result<ModuleTwin, AzureError>
+    where
+        S: Into<String>,
+        T: Into<String>,
+    {
+        let uri = format!(
+            "https://{}.azure-devices.net/twins/{}/modules/{}?api-version={}",
+            self.iothub_name,
+            device_id.into(),
+            module_id.into(),
+            API_VERSION
+        );
+
+        get_twin(self, uri).await
```

Comment:
It seems a bit error prone that `get_twin` takes an arbitrary string that it hopes is the right URI.

Reply:
That's fair, I moved building URI to `get_twin` instead.